generators/app/tasks/bitriseInitialization.js

```diff
@@ -0,0 +1,112 @@
+const runCommand = require('./runCommand');
+const createBitriseApp = require('./createBitriseApp');
+const fs = require('fs');
+
+module.exports = function bitriseInitialization() {
+    return this.prompt([
+        {
+          type: 'confirm',
+          name: 'bitrise',
+          message: 'Do you want to configure bitrise?'
+        }
+      ]).then(({ bitrise }) => {
+          if(bitrise){
+            return runCommand({
+                command: ['ssh-keygen', ['-t', 'rsa', '-b', '4096', '-P', '', '-f', './bitrise-ssh', '-m', 'PEM']],
+                loadingMessage: 'Creating ssh key...',
+                context: this.options
+              }).then(({ spinner }) =>
+              {
+                spinner.stop();
+                return this.prompt([
+                    {
+                      type: 'input',
+                      name: 'repoUrl',
+                      message: "What's your repository url? (ssh only)",
+                      validate: val => (val ? true : 'Repository url (ssh) is required to configure bitrise')
+                    }
+                  ]).then(({ repoUrl }) => {
+                    this.repoUrl = repoUrl;
+                    // git remote add origin <url>
+                    return this.prompt([
+                        {
+                          type: 'input',
+                          name: 'isPublic',
+                          message: "Is your repo public? If true then the repository visibility setting will be public, in case of false it will be private",
+                          validate: val => (val ? true : 'This value is required to configure bitrise')
+                        }
+                      ]).then(({ isPublic }) => {
+                        this.isPublic = isPublic;
+                        return this.prompt([
+                            {
+                              type: 'input',
+                              name: 'repoSlug',
+                              message: "Write the repo slug (The name of your repo not the url)",
+                              validate: val => (val ? true : 'Repository slug is required to configure bitrise')
+                            }
+                          ]).then(({ repoSlug }) => {
+                            this.repoSlug = repoSlug;
+                            return this.prompt([
+                                {
+                                  type: 'input',
+                                  name: 'gitOwner',
+                                  message: "Who is the owner of the repo?",
```

Comment:
This is really required? 🤔

Reply:
Bitrise api wants that info. I suppose that it use it to send some stuff to the git api by itself